apps/web/utils/ai/example-matches/find-example-matches.ts

```
@@ -0,0 +1,112 @@
+  });
+
+  const findExampleMatchesToolCalls = aiResponse.toolCalls.filter(
+    ({ toolName }) => toolName === FIND_EXAMPLE_MATCHES,
+  );
+
+  const matches = findExampleMatchesToolCalls.reduce<
+    z.infer<typeof findExampleMatchesSchema>["matches"]
+  >((acc, { args }) => {
+    const typedArgs = args as z.infer<typeof findExampleMatchesSchema>;
+
+    return [...acc, ...typedArgs.matches];
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)